### PR TITLE
Feat distutils configuration

### PR DESCRIPTION
**Pull request by samhanic**
Hello,
First of all, thanks for the work, kneed is very useful and works reliably.

I reworked the packaging system in this PR mainly to resolve the issue #121 (about Matplotlib installation which was compulsory).

This is done by using setuptools optional dependencies features, using `setup.cfg` and `pyproject.toml` configuration file. This way, packaging is up to date with recent setuptools versions.

It also implies dropping 3.6 support, but it should not cause many issues as this version reached its end-of-life since 2021.

### CODE REVIEW

**Comment by arvkevi** [comment]
Thanks so much for the pull request @samhanic, this will be an impactful feature. I will take a closer look later this week.

File: README.md [hunk 1]
```diff
@@ -29,13 +29,14 @@ $ conda install -c conda-forge kneed
...
 ```bash
-$ pip install kneed
+$ pip install kneed # To install only knee-detection algorithm
+$ pip install kneed[plot] # To also install plotting functions for quick visualizations
 ```
 
 **Clone from GitHub**
 ```bash
 $ git clone https://github.com/arvkevi/kneed.git
-$ python setup.py install
```
Comment (arvkevi) [inline README.md:38]: I never caught this, should probably add a line here to change into the kneed directory.  
`cd kneed`

**Review by arvkevi — CHANGES_REQUESTED** [review]
Hi @samhanic, this is looking good. Can you please update `CONTRIBUTING.md`. If we leave it as-is, I think the instructions under "Forking the Repository", list item 3 would be insufficient to start contributing. (we could just simplify the instructions to be "pip install -e .[testing]".

File: README.md (hunk 1 above)
Comment (samhanic) [reply README.md:38]: Yes, I did not caught it but it is quite obvious this is a typo. 

For the CONTIBUTING.md change, I fixed item list 3 as you requested but probably the item 2 will need some fixes too in future PR (the venv activation command is missing)

**Review by arvkevi — APPROVED** [review]

**Pull request merged**